docs/dev/testing.md

```diff
@@ -0,0 +1,44 @@
+### Testing your code
+There are two types of tests in this project. Unit tests that are written in Scala, and integration tests that are written in Python. 
+We encourage writing integration tests if you have to choose between the two as that helps us ensure the plugin is working across different platforms. 
+
+#### Unit tests
+Unit-tests are located [here](tests). In order to run the unit-tests follow these steps
+1. Issue the Maven command to run the tests with `mvn test`. This will run all the tests in the tests module. 
+2. To run individual tests append `-DwildcardSuites=<comma separated list of wildcard suite names to execute>` to the above command 
+
+For more information about using scalatest with maven please refere [here](https://www.scalatest.org/user_guide/using_the_scalatest_maven_plugin)
+    
+##### Running unit-tests against specific Apache Spark versions. 
+There are Maven profiles that should be used to run Apache Spark version-specific tests e.g. To run tests against Apache Spark 3.1.0, 
+`mvn -P spark310tests test -DwildcardSuites="com.nvidia.spark.rapids.ParquetWriterSuite"`. 
+Please refer to the [pom.xml](tests/pom.xml) to see the list of profiles supported
+
+Apache Spark specific configurations can be passed in by setting environment-variable SPARK_CONF 
+
+Examples: 
+- To pass Apache Spark configs `--conf spark.dynamicAllocation.enabled=false --conf spark.task.cpus=1` do something like.
+`SPARK_CONF="spark.dynamicAllocation.enabled=false,spark.task.cpus=1" mvn ...`
+- To run test ParquetWriterSuite in package com.nvidia.spark.rapids, issue `mvn test -DwildcardSuites="com.nvidia.spark.rapids.ParquetWriterSuite"`
+
+
+#### Integration tests
```

Comment:
so there is already documentation in integration_tests.md do we really need this in both places?  or just have a link and put the details in that doc

Reply:
I would prefer to have us link to it.  There is also documentation about unit tests 

https://github.com/NVIDIA/spark-rapids/blob/branch-0.3/docs/testing.md

Not sure why we are not linking to that primarily and updating it as needed.

Reply:
I will link to the testing doc and update if needed. I honestly wasn't aware we have a document already, thanks for bringing it up

Reply:
I'm still confused, why do we have so many different docs for testing. I realize you linked them but now there is info in testing.md and integrationtest.md . We added information here on running but I need to go to integration tests readme.md to know what i need to install to add it.  why don't we put all information in the integration test readme and test readme and just have small description and point to those?   Maybe i'm missing something @revans2 @jlowe opinions?